libs/cardano-ledger-core/src/Cardano/Ledger/Address.hs

```diff
@@ -791,13 +808,24 @@ decodeVariableLengthWord32 name buf = do
         res <- decode7BitVarLength name buf (\_ -> failDecoding name "too many bytes.") acc
         -- Only while decoding the last 7bits we check if there was too many
         -- bits supplied at the beginning.
-        unless (bufUnsafeIndex buf off0 .&. 0b01110000 == 0) $
+        unless (bufUnsafeIndex buf off0 .&. 0b11110000 == 0b10000000) $
```

Comment:
Change to this check is not needed, this is mostly to clarify that the most significant bit is always set. In other words it is impossible for it not to be set at this particular point.